pvlib/iotools/pvgis.py

```diff
@@ -510,6 +536,16 @@ def get_pvgis_tmy(latitude, longitude, outputformat='json', usehorizon=True,
     if map_variables:
         data = data.rename(columns=VARIABLE_MAP)
 
+    if roll_utc_offset is not None or coerce_year is not None:
+        # XXX: be explicit, test for identity not implicit booleaness
+        # roll_utc_offset None but coerce_year isn't, set year with utc zero
+        if roll_utc_offset is None:  # XXX: None and zero are both False
+            roll_utc_offset = 0
```

Comment:
Is this really a desired behavior? Alternatively would be to raise a ValueError if roll_utc_offset is missing

Reply:
I think it's important because it allows user to coerce year, but leave TZ alone - especially useful in GMT. @kandersolar requested this in https://github.com/pvlib/pvlib-python/pull/2138#discussion_r1696869191. I reworked `_coerce_and_roll_tmy()` to accept either `None` or zero as UTC, so I removed this from `get_pvgis_tmy()` for simplicity.

Reply:
Done in fb2813d.

Reply:
I guess a user could pass in a non-utc localized index (assuming the timezone would get picked up by the function) but then it would get overridden if the timezone wasn't separately given as an input?

Reply:
`_coerce_and_roll_tmy()` is a private function in the `pvgis.py` module that only gets called by `get_pvgis_tmy()`, and PVGIS TMY are always UTC. This is the purpose of #2139 which is to localize TMY dataset from PVGIS and roll it so it starts at midnight.

IMHO private functions are undocumented, so responsibility is the at the user’s own risk if `_coerce_and_roll_tmy()` is passed a localized index.

Reply:
Right right, I'm not requesting changes just wanted to clarify my understanding.